src/pages/Courses/utils/getFilteredData.js

```diff
@@ -0,0 +1,41 @@
+export const getFilteredData = (state, productList) => {
+  const { minRating, minPrice, category, sortType } = state;
+
+  // sort by price
+
+  let sortedProductList = [
+    ...productList.sort((product1, product2) => {
+      if (sortType == "ascending") {
+        return product1.discountedPrice - product2.discountedPrice;
+      } else if (sortType == "descending") {
+        return product2.discountedPrice - product1.discountedPrice;
+      }
+    }),
+  ];
+
+  // filter by rating
+
+  let filteredData = [
+    ...sortedProductList.filter((product) => product.rating >= minRating),
+  ];
```

Comment:
Suggestion: filter method does give a brand new array, so no need to spread it deliberately